scripts/irbrc.rb

```diff
@@ -73,3 +38,40 @@
 rescue => exception
   puts "Error : 'load #{ENV["HOME"]}/.irbrc' : #{exception.message}"
 end
+
+# Turn on completion.
+begin
+  require "readline"
+
+  require "irb/completion" rescue nil
+
+  # Turn on history saving.
+  # require "irb/ext/save-history"
+  # IRB.conf[:HISTORY_FILE] = File.join(ENV["HOME"], ".irb-history")
+
+  # Use an alternate way to on history saving until save-history is fixed.
+  #
+  #   bug:   http://redmine.ruby-lang.org/issues/show/1556
+  #   patch: http://pastie.org/513500
+  #
+  # This technique was adopted from /etc/irbrc on OS X.
+  histfile = File.expand_path(".irb-history", ENV["HOME"])
+  histfile = IRB.conf[:HISTORY_FILE] || histfile
```

Comment:
Based on http://stackoverflow.com/questions/2065923/irb-history-not-working It seems that there was an error in Ruby 1.8. As of now it should be enough to use the line which @mpapis proposed. 

@mpapis why haven't we patched 1.8 directly and instead we fix it in irbrc.rb?